src/mesh/BoundingBox.cpp

```diff
@@ -0,0 +1,171 @@
+#include "BoundingBox.hpp"
+
+namespace precice {
+namespace mesh {
+
+BoundingBox::BoundingBox(std::vector<double> bounds)
+{
+  //PRECICE_CHECK(bounds.size() == 4 || bounds.size() == 6, "Dimension of the bounding box should be 2 or 3.");
+  _bounds = bounds;
+  _dimensions = _bounds.size() / 2;
+}
+
+BoundingBox::BoundingBox(int dimension)
+:_dimensions(dimension)
+{
+  for(int i = 0; i < _dimensions; ++i){
+    _bounds.push_back(std::numeric_limits<double>::max());
+    _bounds.push_back(std::numeric_limits<double>::lowest());
+  }
+}
+
+BoundingBox::BoundingBox()
+{
+  _dimensions = 3;
+  for(int i = 0; i < _dimensions; ++i){
+    _bounds.push_back(std::numeric_limits<double>::max());
+    _bounds.push_back(std::numeric_limits<double>::lowest());
+  }
+}
+
+BoundingBox::BoundingBox(const BoundingBox &bb)
+{
+  _dimensions = bb._dimensions;
+  _bounds     = bb._bounds;
+}
+
+BoundingBox::~BoundingBox()
+{
+}
+
+std::ostream &operator<<(std::ostream &out, const BoundingBox &bb)
+{
+  for (int d = 0; d < bb._dimensions; ++d) {
+    out << "dim: " << d << " min: " << bb._bounds[2 * d] << ", max: " << bb._bounds[2 * d + 1] << "\n";
+  }
+  return out;
+}
+
+bool BoundingBox::operator==(const BoundingBox& otherBB) const
+{
+  if(_dimensions != otherBB._dimensions){
+    return false;
+  }
+  for(int i = 0; i < _dimensions; ++i){
+    if(std::abs(_bounds.at(i) - otherBB._bounds.at(i)) > 1e-6){
+      return false;
+    }
+  }
+  return true;
+}
+
+void BoundingBox::modifyForTest(int rank, std::string testName)
+{
+  if(testName == "com")
+  {
+    for (int i = 0; i < _dimensions; i++) {
+      _bounds[2*i] = rank*i;
+      _bounds[2*i + 1] = i + 1;
+    }
+  }
+  if(testName == "partition"){
+    for (int i = 0; i < _dimensions; i++) {
+      _bounds[2*i] = 3 - rank - 1;
+      _bounds[2*i + 1] = 3 - rank;
+    }
+  }
+}
+
+BoundingBox BoundingBox::createFromData(std::vector<double> bounds)
+{
+  BoundingBox box{bounds};
+  return box;
+}
+
+void BoundingBox::setSafetyFactor(double safetyFactor){
+  _safetyFactor = safetyFactor;
+}
+
+bool BoundingBox::isVertexInBB(const mesh::Vertex &vertex) const
+{
+  for (int d = 0; d < _dimensions; d++) {
+    if (vertex.getCoords()[d] < _bounds.at(2 * d) || vertex.getCoords()[d] > _bounds.at(2 * d + 1)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+double BoundingBox::getData(int dimension, int type) const
+{
+  return _bounds.at(2*dimension + (type - 1));
+}
```

Comment:
I think this should not be needed. All functionality could be move to `BoundingBox`, such as `Mesh::getCOG`.